plugin/storage/badger/spanstore/writer.go

```diff
@@ -128,6 +142,22 @@ func createIndexKey(indexPrefixKey byte, value []byte, startTime uint64, traceID
 	return key
 }
 
+func createOperationWithKindIndexKey(value []byte, startTime uint64, traceID model.TraceID, kind model.SpanKind) []byte {
+	// KEY: indexKey<indexValue><startTime><spanKind><traceId> (traceId is last 16 bytes of the key)
+	key := make([]byte, 1+len(value)+8+1+sizeOfTraceID)
+	key[0] = operationNameWithKindIndexKey
+	pos := len(value) + 1
+	copy(key[1:pos], value)
+	binary.BigEndian.PutUint64(key[pos:], startTime)
+	pos += 8
+	key[pos] = getBadgerSpanKind(kind).String()[0]
```

Comment:
It looks like you're reading the first character of the string - this is a bad pattern. Please define the enum as alias to `byte` type and explicitly assign letters.